solidity/contracts/lib/zeto_base.sol

```diff
@@ -62,14 +81,38 @@ abstract contract ZetoBase is IZetoBase, ZetoCommon {
             if (i > 0 && sortedInputs[i] == sortedInputs[i - 1]) {
                 revert UTXODuplicate(sortedInputs[i]);
             }
-            if (_utxos[sortedInputs[i]] == UTXOStatus.UNKNOWN) {
+            if (
+                _lockedUtxos[sortedInputs[i]] == UTXOStatus.UNKNOWN &&
+                _utxos[sortedInputs[i]] == UTXOStatus.UNKNOWN
+            ) {
                 revert UTXONotMinted(sortedInputs[i]);
-            } else if (_utxos[sortedInputs[i]] == UTXOStatus.SPENT) {
+            }
+            if (
+                _lockedUtxos[sortedInputs[i]] == UTXOStatus.SPENT ||
+                _utxos[sortedInputs[i]] == UTXOStatus.SPENT
+            ) {
```

Comment:
Reading the logic, it feels like:
we allow the same UTXO hash to exist in both the regular and locked UTXO tracking map.
but once one of the UTXO with that hash is spent, the other UTXO will not be spendable.

@jimthematrix did I miss a uniqueness check somewhere

Reply:
we don't allow the same UTXO to exist in both maps at the same time. if you think that's a possibility due to a hole in the current validation logic, please provide the details

Reply:
At the first read, it wasn't clear to me the whether someone can use the lock function to generate same utxos. I added a test to prove that's not possible: https://github.com/hyperledger-labs/zeto/pull/121/commits/8ced077be78dc883e572a9bd90c3a2664294275c